### PR TITLE
Feature/jax and numba

### PR DESCRIPTION
**Pull request by Jammy2211**
In this PR, I have begun to adapt the source code under the assumption that both `jax` and `numba` are installed and working.

I have then been making whatever changes are necessary to make unit tests pass, with the majority of updates making calculations (which I believe are not used in a likelihood function) use normal ndarrays and often `numba`.

I have been doing this alongside the `jax.grad` of an autogalaxy likelihood function, and along the way caught a few issues where I need to be careful which parts of the code assume only normal `numpy` and which parts require careful treatment of whether an array is an `np.ndarray` or `jnp.ndarray`. 

This has led to a few `if isinstance(array, jnp.ndarray)` statements, especially in parts of the code where both an `np.ndarray` or a `jnp.ndarray` could go through.

Overall, it seems to be going smoothly and makes me optimistic we can reach a point where all unit tests pass and normal science runs are working without too much more time and effort, albeit a lot of work will then need to go towards cleaning up the code down the line :).

### CODE REVIEW

File: autoarray/geometry/geometry_util.py [hunk 1]
```diff
@@ -751,30 +747,16 @@ def grid_pixel_centres_2d_from(
...
-                    (grid_scaled_2d[y, x, 1] / pixel_scales[1]) + centres_scaled[1] + 0.5
-                )
-
-    return grid_pixels_2d
+    centres_scaled = np.array(centres_scaled)
+    pixel_scales = np.array(pixel_scales)
+    sign = np.array([-1.0, 1.0])
+    return (
+        (sign * grid_scaled_2d / pixel_scales) + centres_scaled + 0.5
+    ).astype(int)
```
Comment (rhayes777) [inline autoarray/geometry/geometry_util.py:755]: Nice so much easier to look at without the dual functionality

File: autoarray/operators/convolver.py [hunk 2]
```diff
@@ -221,7 +221,7 @@ def __init__(self, mask, kernel):
                         coordinates=(x, y),
                         mask=np.array(mask),
                         mask_index_array=self.mask_index_array,
-                        kernel_2d=np.array(self.kernel.native[:, :]),
+                        kernel_2d=self.kernel.native,
```
Comment (rhayes777) [inline autoarray/operators/convolver.py:224]: Ha that's suprising?

File: autoarray/operators/over_sampling/over_sample_util.py [hunk 3]
```diff
@@ -364,11 +362,11 @@ def sub_size_radial_bins_from(
...
             if radial_grid[i] < radial_list[j]:
-                if use_jax:
-                    # while this makes it run, it is very, very slow
-                    sub_size = sub_size.at[i].set(sub_size_list[j])
-                else:
-                    sub_size[i] = sub_size_list[j]
+                # if use_jax:
+                #     # while this makes it run, it is very, very slow
+                #     sub_size = sub_size.at[i].set(sub_size_list[j])
+                # else:
```
Comment (rhayes777) [inline autoarray/operators/over_sampling/over_sample_util.py:368]: Commented code?

File: test_autoarray/mask/test_mask_1d.py [hunk 4]
```diff
@@ -53,34 +51,34 @@ def test__constructor__input_is_2d_mask__raises_exception():
...
     mask = aa.Mask1D(mask=[True, True, True, True], pixel_scales=1.0)
 
-    assert mask.is_all_true is True
+    assert mask.is_all_true == True
 
 
 def test__is_all_false():
     mask = aa.Mask1D(mask=[False, False, False, False], pixel_scales=1.0)
 
-    assert mask.is_all_false is True
```
Comment (rhayes777) [inline test_autoarray/mask/test_mask_1d.py:74]: is is correct
